esx_service/utils/auth.py

```diff
@@ -84,7 +149,18 @@ def get_tenant(vm_uuid):
             tenant_name = result[0]
             logging.debug("Found tenant_uuid %s, tenant_name %s", tenant_uuid, tenant_name)
 
-    return None, tenant_uuid, tenant_name
+        return None, tenant_uuid, tenant_name
+    else:
+        error_info, tenant_uuid, tenant_name = get_fallback_tenant()
```

Comment:
get_fallback_tenant() seems to be doing pretty much the dame as the code above (line 136-151). I think it would be better wrap it in a function get_tenant_from_db(where_clause, key) , and use it with something like.
```python
get_tenant_from_db("id = ?", (tenant_uuid, ))
```

Basically I think copy-n-paste is evil :-)